src/app/personside/infotabs/meldinger/traadliste/TraadListeElement.tsx

```diff
@@ -112,4 +112,4 @@ function TildeltSaksbehandlerEtikett({ traadId }: { traadId: string }) {
     return null;
 }
 
-export default TraadListeElement;
+export default React.memo(TraadListeElement);
```

Comment:
Denne komponenten er vel ikke pure?
Vil den ikke få trøbbel med at endringer i redux-state ikke blir fanget opp?

Reply:
Har selvfølgelig testet mens jeg jobbet, ingen uventet oppførsel. Har ikke slengt rundt meg med memo men prøvd å begrense det til noen få steder hvor det ga mye verdi. Noe spesielt du tenker på?

Reply:
Bakgrunnen for kommentaren; 
> If your function component renders the **same result given the same props**, you can wrap it in a call to React.memo for a performance boost in some cases by memoizing the result. This means that React will skip rendering the component, and reuse the last rendered result.
https://reactjs.org/docs/react-api.html#reactmemo

Testet lokalt og komponenter ble rerenderet innenfor `memo` om de brukte `useContext`. 
Bekymringen er dog om dette er å regne som en udokumentert "feature" som pluteslig forsvinner i neste react-versjon. 
Alt alt ett issue på å få dette dokumentert bedre fra react-teamet; så gi gjerne ett liten thumbs-up der.
https://github.com/reactjs/reactjs.org/issues/1923

Reply:
Aight! Har testet jeg og, ser jo ut til å gi den oppførselen vi er ute etter. Kan vel også hende at useSelector fra react redux gjør noe magi i denne sammenhengen. Vi får følge med 👍

Reply:
`useSelector` hjelper til med å hindre rerender: https://github.com/reduxjs/react-redux/blob/master/src/hooks/useSelector.js#L76

Men det potensielle issuet er at `useContext()` ikke lenger fører til rerenders i det hele tatt. 
Var visse tilfeller med react@15 der dette kunne skje, men i react@16 og det nye context-apiet ser det ut som meningen er at oppdatering til context alltid skal føre til rerendering uavhengig av scu/pureComponent/memo. Slik som det er nevnt i det issuet linket til ovenfor: 

> The propagation from Provider to its descendant consumers is not subject to the shouldComponentUpdate method, so the consumer is updated even when an ancestor component bails out of the update.